### ISSUE TITLE
Mood Semantic Effects: Learn modality meaning

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#103]
## Overview

Mood (indicative/conditional/imperative/infinitive) is partially captured in the AST but carries significant **semantic meaning** about reality, hypotheticals, and speaker intent.

## Current State

**In AST**: INCONSISTENT
```python
"us": {"vortspeco": "verbo", "tempo": "kondiĉa"},  # Uses 'tempo' but it's mood\!
"u": {"vortspeco": "verbo", "modo": "imperativo"},
"i": {"vortspeco": "verbo", "modo": "infinitivo"},
```

**Problems**:
1. Conditional (-us) stored as `tempo` instead of `modo`
2. Indicative (default) not explicitly marked
3. Semantic effect not learned

### Modality Types
- **Indicative** (-as/-is/-os): Asserts reality, factual claims
- **Conditional** (-us): Hypothetical, counterfactual, uncertain
- **Imperative** (-u): Directive, wish, command
- **Infinitive** (-i): Non-finite, potential action

### Examples
- "Li venas" (He comes) - fact about reality
- "Li venus" (He would come) - hypothetical, NOT real
- "Li venu" (Let him come) - wish/command
- "Li volas veni" (He wants to come) - potential action

### Critical for Reasoning
- "Se mi estus riĉa, mi vojaĝus" = explicitly NOT about reality
- Conditional sentences should NOT be treated as facts
- Commands express desire, not truth claims

### 1. Fix AST Structure (Issue #91)
```python
# All moods should use 'modo' field
"us": {"vortspeco": "verbo", "modo": "kondicionalo"},
"u": {"vortspeco": "verbo", "modo": "imperativo"},
"i": {"vortspeco": "verbo", "modo": "infinitivo"},
"as": {"vortspeco": "verbo", "tempo": "prezenco", "modo": "indikativo"},
```

### 2. Semantic Effect (Learned)
```python
class MoodTransformer(nn.Module):
    def __init__(self, dim):
        self.mood_transforms = nn.ModuleDict({
            'indikativo': nn.Identity(),  # Baseline
            'kondicionalo': nn.Linear(dim, dim),  # Hypothetical
            'imperativo': nn.Linear(dim, dim),    # Command
            'infinitivo': nn.Linear(dim, dim),    # Potential
        })
    
    def forward(self, embedding, mood):
        return self.mood_transforms[mood](embedding)
```

### 3. Training Pairs
```python
# Indicative vs Conditional - factual vs hypothetical
("Li venas", "Li venus", similarity=0.3)  # Very different\!

# Conditional pairs should cluster together
("Mi venus", "Li venus", similarity=0.8)  # Same mood

# Commands vs statements
("Venu\!", "Li venas", similarity=0.4)
```

### 4. Evaluation Metrics
| Metric | Target |
|--------|--------|
| Mood classification accuracy | >95% |
| Factual vs hypothetical discrimination | >80% |
| Command detection | >90% |

## Dependencies

- #91 Fix mood/tense structure (structural prerequisite)

## Related Issues

- #101 Meta: Grammatical Features with Semantic Content (parent)
- #91 Mood/tense structure (fix AST first)
- #87 Sentence type (related - commands)
